test/RToken.test.ts

```diff
@@ -1427,120 +1392,63 @@ describe(`RTokenP${IMPLEMENTATION} contract`, () => {
-      expect((await token3.balanceOf(addr1.address)).sub(before3).div(expectedTkn3)).to.equal(28)
-
-      // Check total asset value did not change
-      expect(await facadeTest.callStatic.totalAssetValue(rToken.address)).to.equal(0)
-
-      // Cancel only the first issuance
-      await expect(rToken.connect(addr1).cancel(1, true))
-        .to.emit(rToken, 'IssuancesCanceled')
-        .withArgs(addr1.address, 0, 1, bn('1e24'))
-    })
-
-    it('Should allow the issuer to rollback minting', async function () {
+    it('Should allow the recipient to rollback minting', async function () {
       const issueAmount: BigNumber = bn('50000e18')
 
       // Provide approvals
-      await token0.connect(addr1).approve(rToken.address, initialBal)
-      await token1.connect(addr1).approve(rToken.address, initialBal)
-      await token2.connect(addr1).approve(rToken.address, initialBal)
-      await token3.connect(addr1).approve(rToken.address, initialBal)
-
-      const [, quotes] = await facade.connect(addr1).callStatic.issue(rToken.address, issueAmount)
-      const expectedTkn0: BigNumber = quotes[0]
-      const expectedTkn1: BigNumber = quotes[1]
-      const expectedTkn2: BigNumber = quotes[2]
-      const expectedTkn3: BigNumber = quotes[3]
+      const [, depositTokenAmounts] = await facade.callStatic.issue(rToken.address, issueAmount)
+      await Promise.all(
+        tokens.map((t, i) => t.connect(addr1).approve(rToken.address, depositTokenAmounts[i]))
+      )
 
       // Issue rTokens
-      await rToken.connect(addr1).issue(issueAmount)
+      await rToken.connect(addr1)['issue(address,uint256)'](addr2.address, issueAmount)
 
-      expect(await token0.balanceOf(addr1.address)).to.equal(initialBal.sub(expectedTkn0))
-      expect(await token1.balanceOf(addr1.address)).to.equal(initialBal.sub(expectedTkn1))
-      expect(await token2.balanceOf(addr1.address)).to.equal(initialBal.sub(expectedTkn2))
-      expect(await token3.balanceOf(addr1.address)).to.equal(initialBal.sub(expectedTkn3))
+      // Check that that underlying was transferred
+      await Promise.all(
+        tokens.map(async (t, i) => {
+          expect(await t.balanceOf(addr1.address)).to.equal(initialBal.sub(depositTokenAmounts[i]))
+        })
+      )
 
       // Check initial state
-      await expectUnprocessedIssuance(addr1.address, 0, {})
-      expect(await rToken.balanceOf(addr1.address)).to.equal(0)
+      await expectUnprocessedIssuance(addr2.address, 0, {})
+      expect(await rToken.balanceOf(addr2.address)).to.equal(0)
 
-      // Cancel with issuer
-      await expect(rToken.connect(addr1).cancel(1, true))
+      // Get initial balances
+      const initialRecipientBals = await Promise.all(tokens.map((t) => t.balanceOf(addr2.address)))
+
+      // Check that the depositor cannot cancel the issuance
+      await expect(rToken.connect(addr1).cancel(1, true)).to.be.revertedWith('out of range')
```

Comment:
@tbrent checking that the depositor cannot cancel the issuance

Reply:
Correct, only the `recipient`. The depositer/issuer is really just a funder and nothing else. `recipient` gets credit for everything.

Reply:
Yup, just below https://github.com/reserve-protocol/protocol/pull/529/files#diff-155f84716d700fd1f15d3481eddf87d954b37e763422b0dc5ffc2924c5993834R1431-R1444